src/widgets/infinite-hits/defaultTemplates.ts

```diff
@@ -1,9 +1,15 @@
 import type { InfiniteHitsComponentTemplates } from '../../components/InfiniteHits/InfiniteHits';
 
 const defaultTemplates: InfiniteHitsComponentTemplates = {
-  empty: 'No results',
-  showPreviousText: 'Show previous results',
-  showMoreText: 'Show more results',
```

Comment:
Is it worth it? We're aiming to deprecate string-based templates anyway, should we spend time and add code for this?

Reply:
I was wondering whether simple templates should also be forced to use a function, it seems heavy handed for something like "show more results", which for those people would always be a string. We can of course add the functionality back after the breaking change has happened though

Reply:
It could be worth having a separate discussion around templates in general in our libraries, yes.

In Autocomplete, all templates are functions. In Recommend there's no Templates API, but each component is a function. Whatever we pick, we should aim for a consistent experience.

Templates as strings are more straightforward, I agree, but accepting multiple types also means we have to handle more use cases internally, leading to more code and more opportunities to miss edge cases. I'm not sure this is worth it, especially since unless you use IE11, you can have really terse code with fat arrow functions.